vizro-core/docs/pages/user-guides/custom-figures.md

```diff
@@ -0,0 +1,108 @@
+# How to create custom figures
+
+This guide explains how to create custom figures, which is useful when you need a figure that reacts to controls and actions.
+
+There are two use cases where you might want to use custom figures:
+
+* You need a figure that isn't available in our pre-defined figure functions [`vizro.figures`](../API-reference/figure-callables.md).
+* You need a figure that doesn't fit into the existing predefined components ([`Graph`][vizro.models.Graph], [`Table`][vizro.models.Table] or [`AgGrid`][vizro.models.AgGrid]).
+
+Use the following approach, which is similar to creation of a [custom chart](../user-guides/custom-charts.md), do the following:
+
+1. Define a function that returns a `Dash` component.
+2. Decorate it with the `@capture("figure")` respectively.
+3. The function must accept a `data_frame` argument (of type `pandas.DataFrame`).
+4. The figure should be derived from and require only one `pandas.DataFrame`. Additional dataframes from other arguments will not react to dashboard components such as `Filter`.
+
+<!-- vale off -->
+The example below shows how to create a custom figure — a list of cards created from a `pandas.DataFrame` where the
+number of cards displayed dynamically adjusts based on a `vm.Parameter`.
+
+!!! example "Dynamic number of cards"
```

Comment:
I really like this example but it's quite advanced because it's doing several complicated things:
* setup of mock data (if you can find a better way to do this using some built in `px.data` dataset then please do so)
* custom CSS
* function returns multiple Dash components
* parameter to control a function

Painful though it may be to write 😅 I think we should have a total of three examples on this page which increase in complexity and show different things:
1. a custom KPI card that basically copies and pastes code from our API reference docs and then modifies a simple thing, like you suggest in https://github.com/mckinsey/vizro/pull/529/files#r1644213316
2. a really simple example that writes a whole new non-KPI card Dash component that reacts to a `Filter` (lmk if you need help thinking of some example) 
3. this multiple cards example

Reply:
Alright, sounds good. On two, we could do a simple html.H3 component that dynamically changes the title based on a filter. I think this would the simplest example, especially if we don't want to have another card example 😄

Reply:
Yeah, I think the challenge is to come up with something that really requires `data_frame` and doesn't require a `dbc.Card` 🤔 (Maybe in the future we will make this no longer a requirement since you might like a component that reacts to a parameter and isn't connected to a dataframe, but let's leave it as required for now anyway.)

e.g. if we show something like `len(data_frame)` then it's probably best shown just in a KPI (or non-KPI) card. So if we did something like `return html.H3(f"Your data has {len(data_frame)} rows")` then maybe it will always look kind of stupid compared to just putting it in a `dbc.Card`. 

So maybe my idea of `Filter` + non-card + `data_frame` required is actually stupid here and will just result in a really weird convoluted example 😅

Reply:
I don't think your idea is _that_ stupid 😄 To have a title that dynamically changes based on a filter/parameter is not that uncommon. I can think of a use-case e.g. where you have a business dashboard and you want to dynamically change the title to reflect the department you are filtering on e.g. "Summary for Department {XXX}".

It does make more sense with other components on the page, but I think adding that simple example with an html.Div might be useful already, just to show that you can return any Dash component not just a dbc.Card 😄

Reply:
Cool, sounds good 😀 The problem with updating text based on the filter value is that the value itself isn't actually available directly, you'd have to do `data_frame[column].something`, which also feels a bit awkward and would maybe be more cleanly done through an action. But probably it's fine just as an example to demonstrate the sort of thing you can do.